ekaterinahubkina/rs-aws-shop-be
# Task 10

### What was done:

## Required tasks (80 points for covering all criteria)

- link to the FE app https://d19abc31vx8vrv.cloudfront.net/
- FE PR https://github.com/ekaterinahubkina/nodejs-aws-shop-react/pull/8
- Product Service API URL https://cdyw64p935.execute-api.eu-central-1.amazonaws.com/prod/
- Cart Service API URL http://ekaterinahubkina-cart-api-prod.eu-central-1.elasticbeanstalk.com/
- BFF Service API URL http://ekaterinahubkina-bff-api-prod.eu-central-1.elasticbeanstalk.com/
**Api Gateway is used as proxy to Elastic Beanstalk** https://x7q4i11dkj.execute-api.eu-central-1.amazonaws.com/

- example of how to call `createProduct` lambda with all needed information: URL, payload, headers, etc.

**Url:  https://x7q4i11dkj.execute-api.eu-central-1.amazonaws.com/product/products (Api Gateway bff url)
OR http://ekaterinahubkina-bff-api-prod.eu-central-1.elasticbeanstalk.com/product/products (EB bff url)
OR https://cdyw64p935.execute-api.eu-central-1.amazonaws.com/prod/products (Product service url)
Method: 'POST'
Body (example): { "title": "test", "description": "test", "price": 100, "count": 7 }**

- example how to call Product Service and Cart Service via BFF Service URL

**Cart service:**
http://ekaterinahubkina-bff-api-prod.eu-central-1.elasticbeanstalk.com/cart/ (EB bff)
https://x7q4i11dkj.execute-api.eu-central-1.amazonaws.com/cart/ (Api Gateway bff)

**!!! Please use basic auth while testing with postman and set the "authorization_token" (username:password base64 encoded) to the local storage when testing with the frontend app. If the user does not exist - they will be created as well as the cart for this user.**

**Product service:**
http://ekaterinahubkina-bff-api-prod.eu-central-1.elasticbeanstalk.com/product/ (EB bff)
https://x7q4i11dkj.execute-api.eu-central-1.amazonaws.com/product/ (Api Gateway bff)

---


![image](https://github.com/user-attachments/assets/5c7e1e82-1ec9-4098-ace7-a6b11d57d39e)

## Application Functionality (**MUST HAVE**)

+ [x] 1. Products representation on Home page should be based on Product Service API.
+ [x] 2. Products are coming from Product DB.
+ [x] 3. Product images are not randomly generated on client side. Product image, same as another product model information should be stored on BE side in Product DB.
+ [x] 4. Products might be created through CSV product file import from client side.
+ [x] 5. Cart might be created with appropriate product set.
+ [x] 6. Auth logic should be in place

## Additional (optional) tasks

  How to test:
  - Get products list
  - Create new product
  - Get products list - result shouldn’t have new product
  - Wait more than 2 minutes
  - Get products list - result should have new product

Score: 100/100
